cinema/tests/test_movie_session_api.py

```diff
@@ -0,0 +1,172 @@
+        self.assertEqual(res.data, serializer.data)
+
+    def test_filter_movie_by_genre(self):
+        movie_without_genres = sample_movie(
+            title="test title",
+            description="test description",
+            duration=100
+        )
+        genre_1 = Genre.objects.create(name="Comedy")
+        genre_2 = Genre.objects.create(name="Drama")
+        genre_3 = Genre.objects.create(name="Fantasy")
+        genre_4 = Genre.objects.create(name="Romance")
+        movie_with_genres = sample_movie()
+        movie_with_genres.genres.add(
+            genre_1,
+            genre_2,
+            genre_3,
+            genre_4,
+        )
+        res = self.client.get(MOVIE_URL, data={"genres": f"{genre_1.id},{genre_2.id},{genre_3.id}"})
+        serializer_movie_with_genres = MovieListSerializer(movie_with_genres)
+        serializer_movie_without_genres = MovieListSerializer(movie_without_genres)
+        self.assertIn(serializer_movie_with_genres.data, res.data)
+        self.assertNotIn(serializer_movie_without_genres.data, res.data)
+
+    def test_filter_movie_by_title(self):
+        movie_test_title = sample_movie(
+            title="test title",
+            description="test description",
+            duration=100
+        )
+        sample_movie()
+
+        res = self.client.get(MOVIE_URL, data={"title": "Groundhog Day"})
+        serializer_movie_default_title = MovieListSerializer(Movie.objects.get(title="Groundhog Day"))
+        serializer_movie_with_test_title = MovieListSerializer(movie_test_title)
+        self.assertIn(serializer_movie_default_title.data, res.data)
+        self.assertNotIn(serializer_movie_with_test_title.data, res.data)
+
+    def test_filter_movie_by_actors(self):
+        movie_without_actors = sample_movie(
+            title="test title",
+            description="test description",
+            duration=100
+        )
+        actor_1 = Actor.objects.create(first_name="Bill", last_name="Murray")
+        actor_2 = Actor.objects.create(first_name="Andie", last_name="MacDowell")
+        movie_with_actors = sample_movie()
+        movie_with_actors.actors.add(
+            actor_1,
+            actor_2,
+        )
+        res = self.client.get(MOVIE_URL, data={"actors": f"{actor_1.id},{actor_2.id}"})
+        serializer_movie_with_actors = MovieListSerializer(movie_with_actors)
+        serializer_movie_without_actors = MovieListSerializer(movie_without_actors)
+        self.assertIn(serializer_movie_with_actors.data, res.data)
+        self.assertNotIn(serializer_movie_without_actors.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        url = detail_url(movie.id)
+        res = self.client.get(url)
+        serializer = MovieDetailSerializer(movie)
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
```

Comment:
The method `test_create_movie_forbidden` is correctly named with the `test_` prefix, ensuring it will be discovered by the test runner. No changes are needed here.